src/cluster_legacy.c

```diff
@@ -7029,9 +7029,29 @@ int clusterCommandSpecial(client *c) {
         clusterDelNode(n);
         clusterDoBeforeSleep(CLUSTER_TODO_UPDATE_STATE | CLUSTER_TODO_SAVE_CONFIG);
         addReply(c, shared.ok);
-    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && c->argc == 3) {
-        /* CLUSTER REPLICATE <NODE ID> */
+    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && (c->argc == 3 || c->argc == 4)) {
+        /* CLUSTER REPLICATE (<NODE ID> | NO ONE)*/
         /* Lookup the specified node in our table. */
+        if (c->argc == 4) {
+            if (0 != strcasecmp(c->argv[2]->ptr, "NO") || 0 != strcasecmp(c->argv[3]->ptr, "ONE")) {
+                addReplySubcommandSyntaxError(c);
+                return 1;
+            }
+            if (nodeIsPrimary(myself)) {
+                addReply(c, shared.ok);
+                return 1;
+            }
+            serverLog(LL_NOTICE, "Stop replication and turning myself into empty primary.");
+            clusterSetNodeAsPrimary(myself);
+            clusterPromoteSelfToPrimary();
+            int empty_db_flags = server.repl_replica_lazy_flush ? EMPTYDB_ASYNC : EMPTYDB_NO_FLAGS;
+            emptyData(-1, empty_db_flags, NULL);
+            /* Reset manual failover state. */
+            resetManualFailover();
```

Comment:
it become a primary, do we need this reset? it has no way to enter the failover check logic. or at least we remove the comment since this line is super easy.
```suggestion
            resetManualFailover();
```

Reply:
I'm not sure what you mean by "no way to enter the failover check logic". This is to abort any in-progress failover (which can happen, right?).

Reply:
i mean, when the node become a primary, it has no way enter this mf handle logic.
```
        if (nodeIsReplica(myself)) {
            clusterHandleManualFailover();
```

Reply:
Sorry. I'm not sure I got you right. If you mean that there is no way to have manual failover to be in progress by the time we come to https://github.com/valkey-io/valkey/blob/bed392bebf6a0110f0cf3e86cf3f2c6845f5437a/src/cluster_legacy.c#L7050, then I would disagree with that. Manual failover could start before we got ```CLUSTER REPLICATE NO ONE```. Also, ```CLUSTER RESET``` does the same: it resets any in-progress failover: https://github.com/valkey-io/valkey/blob/bed392bebf6a0110f0cf3e86cf3f2c6845f5437a/src/cluster_legacy.c#L1298

Reply:
i know it could start before the NO ONE, i mean, if mf is in progress, will it be a problem if it is not reset? What will happen? I know CLUSTER RESET do the same thing, it is a old time stuff, i was just thinking about this, we definitely have other places where we haven't reset it when the role changed.

Reply:
> we definitely have other places where we haven't reset it when the role changed.

Interesting. Is this a problem?

Reply:
What is the reason of not resetting it?

Reply:
i am not against it, this is just a question, after it becomes a primary, will there be any problems with the uncleaned mf failover? As far as i know there is no problem (only a log issue like the replica (empty primary) will print the mf timeout something like that). The original primary has a known issue with pause timeout.

i am ok with resetting it. I would probably prefer to have one that refuses to execute NO ONE if the replica is in failover or is in mf failover. The admin should do it in a stable state, which means the replica should not be in the failover state (or at least should not in mf failover). But there is no harm i guess (i dont see a problem here), so i am also ok with it.